recipe/build.sh

```diff
@@ -7,30 +7,66 @@ set -ex
 export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$BUILD_PREFIX/lib/pkgconfig
 export PKG_CONFIG=$BUILD_PREFIX/bin/pkg-config
 
+meson_config_args=(
+    -Dbenchmark=disabled
+    -Dcairo=enabled
+    -Dcoretext=auto
+    -Ddocs=disabled
+    -Dfontconfig=enabled
+    -Dfreetype=enabled
+    -Dgdi=auto
+    -Dglib=enabled
+    -Dgobject=enabled
+    -Dgraphite=enabled
+    -Dicu=enabled
+    -Dintrospection=enabled
+    -Dtests=disabled
+)
+
 if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
-  MESON_ARGS="${MESON_ARGS} -Dintrospection=disabled"
-else
-  MESON_ARGS="${MESON_ARGS} -Dintrospection=enabled"
+  unset _CONDA_PYTHON_SYSCONFIGDATA_NAME
+  (
+    mkdir -p native-build
+
+    export CC=$CC_FOR_BUILD
+    export CXX=$CXX_FOR_BUILD
+    export OBJC=$OBJC_FOR_BUILD
+    export AR="$($CC_FOR_BUILD -print-prog-name=ar)"
+    export NM="$($CC_FOR_BUILD -print-prog-name=nm)"
+    export LDFLAGS=${LDFLAGS//$PREFIX/$BUILD_PREFIX}
+    export PKG_CONFIG_PATH=${BUILD_PREFIX}/lib/pkgconfig
+
+    # Unset them as we're ok with builds that are either slow or non-portable
+    unset CFLAGS
+    unset CPPFLAGS
+    export host_alias=$build_alias
+    export PKG_CONFIG_PATH=$BUILD_PREFIX/lib/pkgconfig
+
+    meson setup native-build \
+        "${meson_config_args[@]}" \
+        --buildtype=release \
+        --prefix=$BUILD_PREFIX \
+        -Dlibdir=lib \
+        --wrap-mode=nofallback
+
+    # This script would generate the functions.txt and dump.xml and save them
+    # This is loaded in the native build. We assume that the functions exported
+    # by glib are the same for the native and cross builds
```

Comment:
Here I think we mean "harfbuzz" not "glib" ... since this is commonly-used boilerplate that gets copy-pasted, should probably make the language more generic, e.g. "We have to assume that the functions exported by the package are the same for the native and cross builds."

Reply:
And I copied it from `pango` too!